homeassistant/components/tplink/switch.py

```diff
@@ -38,9 +39,48 @@ async def async_setup_entry(
     else:
         entities.append(SmartPlugSwitch(device, coordinator))
 
+    entities.append(SmartPlugLedSwitch(device, coordinator))
```

Comment:
Do all devices have leds ?

Reply:
If not we should only create the entity for ones that do

Reply:
All [`is_plug`](https://github.com/python-kasa/python-kasa/blob/master/kasa/smartplug.py#L66) and [`is_strip`](https://github.com/python-kasa/python-kasa/blob/master/kasa/smartstrip.py#L134) do, according to `python-kasa`, which is tested for a few lines above!